src/Infrastructure/Persistence/Contexts/BaseDbContext.cs

```diff
@@ -93,8 +93,8 @@ private List<AuditTrail> OnBeforeSaveChanges(in Guid userId)
     {
         ChangeTracker.DetectChanges();
         var trailEntries = new List<AuditTrail>();
-        foreach (var entry in ChangeTracker.Entries<AuditableEntity>()
-            .Where(e => e.State is EntityState.Added or EntityState.Deleted or EntityState.Modified)
+        foreach (var entry in ChangeTracker.Entries()
+            .Where(e => typeof(IAuditableEntity).IsAssignableFrom(e.Entity.GetType()) && e.State is EntityState.Added or EntityState.Deleted or EntityState.Modified)
```

Comment:
I think this is the same as

`foreach (var entry in ChangeTracker.Entries<IAuditableEntity>().Where(e => e.State...`

Reply:
I thought so, but did not work